src/win/fs.c

```diff
@@ -1701,21 +1702,43 @@ static void fs__symlink(uv_fs_t* req) {
   WCHAR* new_pathw = req->fs.info.new_pathw;
   int flags = req->fs.info.file_flags;
   int result;
+  int err;
 
 
   if (flags & UV_FS_SYMLINK_JUNCTION) {
     fs__create_junction(req, pathw, new_pathw);
-  } else if (pCreateSymbolicLinkW) {
-    result = pCreateSymbolicLinkW(new_pathw,
-                                  pathw,
-                                  flags & UV_FS_SYMLINK_DIR ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0) ? 0 : -1;
-    if (result == -1) {
-      SET_REQ_WIN32_ERROR(req, GetLastError());
-    } else {
-      SET_REQ_RESULT(req, result);
-    }
-  } else {
+    return;
+  }
+  if (!pCreateSymbolicLinkW) {
     SET_REQ_UV_ERROR(req, UV_ENOSYS, ERROR_NOT_SUPPORTED);
+    return;
+  }
+
+  if (flags & UV_FS_SYMLINK_DIR) {
+    flags = SYMBOLIC_LINK_FLAG_DIRECTORY;
+  } else {
+    flags = uv__file_symlink_usermode_flag;
+  }
+
+  result = pCreateSymbolicLinkW(new_pathw, pathw, flags);
+  if (result != 0) {
+    SET_REQ_RESULT(req, 0);
+    return;
+  }
+
+  /* Something went wrong. We will test if it is because of user-mode
+   * symlinks.
+   */
+  err = GetLastError();
+  if (err == ERROR_INVALID_PARAMETER &&
+      flags & SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE) {
+    /* This system does not support user-mode symlinks. We will clear the
+     * unsupported flag and retry.
+     */
+    uv__file_symlink_usermode_flag = 0;
+    fs__symlink(req);
```

Comment:
Is it hypothetically possible for `fs__symlink()` to keep calling itself? Maybe add a check that `uv__file_symlink_usermode_flag` isn't already zero?